go/pserver/etcd_lister.go

```diff
@@ -0,0 +1,116 @@
+package pserver
+
+import (
+	"time"
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/coreos/etcd/clientv3"
+	log "github.com/sirupsen/logrus"
+)
+
+const DefaultEtcdTimeout time.Duration = time.Second * time.Duration(5)
+
+type pserverEtcdLister struct {
+	client    *clientv3.Client
+	timeout   time.Duration
+	endpoints []string
+}
+
+// read ps desired number from etcd.
+func(p pserverEtcdLister) desired() int {
```

Comment:
Can put https://github.com/PaddlePaddle/Paddle/blob/develop/go/pserver/service.go#L106 as a function and call it directly.